### DISCUSSION TITLE
Nuxt 3 bundled entry.mjs getting bigger and bigger in size

### DISCUSSION THREAD

**Discussion by tiborpino** [nuxt/nuxt#16527] [Questions]
Hi there. Our dist/client/entry-*.mjs is getting bigger and bigger as our codebase getting bigger. Is there an option to make improvements in bundle size?

I've tried lazy loaded components but it had no impact.

Thank you

**Comment by tkjaergaard** [comment]
@tiborpino I'm experiencing the same thing.

@pi0 any hints? 🙂

**Comment by bhaskarGyan** [comment]
You can follow below approach based on the type of library you are adding.

- **Component Lib** (Like Chartsjs,vue-chartjs vue3-carousel etc)
    Create your component wrapper (https://github.com/bhaskarGyan/movies/blob/lazy-component/components/VCarousel.vue#L1) 
    add `lazy-` while using the component in Pages or within other components (https://github.com/bhaskarGyan/movies/blob/lazy-component/pages/playground.vue#L18). 
    It will create a lazy chunk for this library (https://movies-hd1mduiss-bhaskargyan.vercel.app/_nuxt/VCarousel.d5ea944d.js)
- **Utils Library** (Like Cryptojs, date-fns etc)
    Do a dynamic import for any heavy utils library (https://github.com/bhaskarGyan/movies/blob/lazy-component/pages/playground.vue#L2)
    it will create a lazy chunk ([index-xxxx.js](https://movies-hd1mduiss-bhaskargyan.vercel.app/_nuxt/index.d307da9c.js)) for the dynamic imported Lib and will be imported on a need basis

**Pages** and the **custom components** (within component folder) are generally lazy loaded as chunks. 

you can check the content of the entry file by running `npx nuxi analyze` and can implement the above two approach based in the type of Lib.

Below is the demo for the above (Same codebase with exactly the same 3rd party component and Lib)

1. [Normal Component large entry js](https://movies-b5il7cmvz-bhaskargyan.vercel.app/playground) - **entry.js -> 75kb**
2. [Lazy Loaded component + dynamic import of util Library](https://movies-hd1mduiss-bhaskargyan.vercel.app/playground) - **entry.js -> 64.9kb**



**GitHub repo**

1. [Normal Component large](https://github.com/bhaskarGyan/movies/tree/normal-component)
2. [Lazy Loaded component + dynamic import of util Library](https://github.com/bhaskarGyan/movies/tree/lazy-component)
